tests.rb

```diff
@@ -17,12 +18,546 @@
+    assert_equal 4, lesson.child_lessons.length
+  end
+
+# End Person A Tests
+
+  # B-Test-1
+  def test_a_reading_is_destroyed_when_its_lesson_is_destroyed
+    lesson_test = Lesson.create(course_id: 99, parent_lesson_id: 99, name: "Test Reading destroyed", pre_class_assignment_id: 1, in_class_assignment_id: 1)
+
+    Reading.create(lesson_id: lesson_test.id, caption: "Testy test", order_number: 66)
+
+    lesson_test.destroy
+    refute Reading.find_by(caption: "Testy test")
+  end
+
+  #B-Test-2
+  def test_destroying_a_course_destroys_its_associated_lessons
+    course_test = Course.create(name: "Destroying lessons like a BAWSS", course_code: "ncc74656")
+
+    Lesson.create(course_id: course_test.id, parent_lesson_id: 99, name: "Destroy this lesson!")
+
+    Lesson.create(course_id: course_test.id, parent_lesson_id: 99, name: "Destroy this lesson too!")
+
+    course_test.destroy
+    refute Lesson.find_by(name: "Destroy this lesson!")
+    refute Lesson.find_by(name: "Destroy this lesson too!")
+  end
+
+  #B-Test-3
+  def test_that_a_course_with_instructors_cannot_be_deleted
+    course_test = Course.create(name: "Destroying lessons like a BAWSS", course_code: "ncc1764")
+    CourseInstructor.create(course_id: course_test.id)
+
+    refute course_test.destroy
+  end
+
+  #B-Test-4
+  def test_that_a_lesson_is_associated_with_its_in_class_assignment
+    assign_test = Assignment.create(name: "Assignment Test", course_id: @course.id, percent_of_grade: 0.34, active_at: Date.today, due_at: "2017-05-15")
+    assert assign_test.persisted?
+    assert assign_test.save!
+
+    lesson_test = Lesson.create(name: "Test In Class Lesson is associated",in_class_assignment: assign_test)
+    assert lesson_test.persisted?
+    assert lesson_test.save!
+    assert lesson_test.in_class_assignment == assign_test
+  end
+
+  #B-Test-5
+  def test_a_course_has_many_readings_through_lessons
+    course_many_readings_test = Course.create(name: "Advanced Lesson Destroying", course_code: "ncc2000")
+
+    lesson_test = Lesson.create(course_id: course_many_readings_test.id, name: "Lesson Destroying Best Practices")
+
+    lesson_test2 = Lesson.create(course_id: course_many_readings_test.id, name: "Lesson Destroying: Safety")
+
+    Reading.create(lesson_id: lesson_test.id, caption: "Lesson Destroying: Industry Methods and Standards", url: "http://destroythelesson.com", order_number: 1)
+
+    Reading.create(lesson_id: lesson_test.id, caption: "How to destroy Lessons Safely", url: "http://destroythelesson.com", order_number: 1)
+
+    Reading.create(lesson_id: lesson_test2.id, caption: "What to do after you've destroyed a lesson",url: "http://destroythelesson.com", order_number: 1)
+
+    assert course_many_readings_test.readings.count > 2
+  end
+
+  #B-Test-6
+  def test_validate_a_school_has_a_name
+    new_school = School.create()
+    assert new_school.name == nil
+    assert new_school.errors.messages
```

Comment:
It won't actually have any errors until you attempt to save it.

This passes because `new_school.errors.messages` is an empty array, which isn't `nil` or `false`